mula/scheduler/clients/http/service.py

```diff
@@ -105,13 +108,31 @@ def post(self, url: str, payload: dict[str, Any], params: dict[str, Any] | None
         Returns:
             A request.Response object
         """
-        response = self.session.post(url, headers=self.headers, params=params, data=payload, timeout=self.timeout)
+        response = self._request_with_backoff(
+            method="POST", url=url, headers=self.headers, params=params, data=payload, timeout=self.timeout
+        )
         self.logger.debug("Made POST request to %s.", url, name=self.name, url=url, payload=payload)
 
         response.raise_for_status()
 
         return response
 
+    def _request_with_backoff(self, method: str, url: str, **kwargs: Any) -> httpx.Response:
+        for i in range(1, self.retries + 1):
+            try:
+                response = self.session.request(method, url, **kwargs)
+                response.raise_for_status()
+                return response
+            except httpx.RequestError as exc:
+                if i == self.retries:
+                    raise exc
+
+                self.logger.warning("Retrying %s request to %s (attempt %d/%d)", method, url, i, self.retries)
+                delay = min(2**i, 60)  # Exponential backoff with a max delay of 60 seconds
+                time.sleep(delay)
+
+        raise RuntimeError("Request failed after maximum retries")
```

Comment:
Do we actually reach this statement?

Reply:
No, but precommit was giving me a hard time about it. And explicitly wanted me to put that there 🤷

Reply:
I think it's a fair complaint from pre-commit/ mypy; I think the return type should be a `httpx.Response | NoReturn`, this way you can leave out the `return` statement if I'm not mistaken